pkg/stats/set_test.go

```diff
@@ -176,7 +176,7 @@ func TestSetHistoryDistribution(t *testing.T) {
 	a.Equal(step(1), [3][]float64{{3, 6, 1}, {3, 6, 1}, {3, 6, 1}})
 	a.Equal(step(2), [3][]float64{{3, 6, 1, 2}, {3, 6, 1, 2}, {3, 6, 1, 2}})
 	a.Equal(step(1), [3][]float64{{3, 1}, {3, 1}, {3, 1}})
-	a.Equal(step(10), [3][]float64{{3, 1, 1}, {3, 1, 10}, {3, 1, 10}})
+	a.Equal(step(10), [3][]float64{{3, 1, 1}, {3, 1, 1}, {3, 1, 10}})
```

Comment:
This change looks suspicious. Why did the test have to change?

Reply:
I don't know.

Reply:
It is not clear what lib behaves correct.
Let's ask Dmitry. I hope he knows more.

Reply:
I don't see any visual artifacts. It should be Ok to accept.

Reply:
What 50-th percentile is the correct one in this case?

Reply:
Here we calculate percentiles per bucket, right?

So by line 177, we have 4 buckets with 1 elements each, therefore all percentiles are equal.
Then, I guess, we start reducing the number of buckets. But how?

At line 178, we have suddenly two buckets (?): `{3,6}` and `{1,2}`. But where did the newly added `1` go? Is it `{3,6}` and `{1,2,1}`? Or, rather `{3,6}`, `{1,2}` and `{1}`? But why are there not 3 elements and we only see 3 elements at line 179?

Reply:
One more detail - new lib has tests, old lib doesn't have tests.

Reply:
> Here we calculate percentiles per bucket, right?

IIRC this is history per percentile.
The 3 arrays correspond to 10/50/90 percentiles. Values in each array is history.

> Then, I guess, we start reducing the number of buckets. But how?

We compress history when it reaches size 4 (newSet arg):
https://github.com/google/syzkaller/blob/master/pkg/stats/set.go#L372-L401

> At line 178, we have suddenly two buckets (?)

History of size 4 is compressed to 2 (half).

> But where did the newly added 1 go?

I guess it just does not represent any of 10/50/90 percentile.

> But why are there not 3 elements and we only see 3 elements at line 179?

When we have 5 samples and compressed 2x, we have 2 complete history elements with 2 samples each, and the last incomplete history element with only 1 sample so far. The last incomplete element is not shown. We need 6 samples to show 3 complete history elements.

Reply:
Thanks!

So, at line 178, we have `{3,6}` and `{1,2}`.
And at line 179, we have `{3,6}`, `{1,2}` and `{1,10}`. 

Technically, `1` should also be a valid median value of `{1,10}`.